integration/e2e_test.go

```diff
@@ -52,6 +52,7 @@ func TestBasic(t *testing.T) {
 
 	deployCluster(t)
 	deployMicrobot(t)
+	upgradeCluster(t)
```

Comment:
Let's run these as subtests:

```suggestion
        t.Run("DeployCluster", deployCluster)
        t.Run("DeployTestWorkload", deployMicrobot)
        t.Run("UpgradeCluster", upgradeCluster)
```

Reply:
for t.Run, the second argument requires a function taking a pointer to testing.T interface, but all these tests function and use testing.TB as an interface rather than a pointer to the testing.T interface.

Reply:
In this case, this should do it:
```
t.Run("DeployCluster", func(t *testing.T) {
   deployCluster(t)
}
```